xrally_kubernetes/tasks/scenario.py

```diff
@@ -0,0 +1,38 @@
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+from rally.common.plugin import plugin
+from rally.common import validation
+from rally.task import scenario
+
+from xrally_kubernetes import service as k8s_service
+
+
+@validation.add_default("required_kubernetes_platform")
+@plugin.default_meta(inherit=False)
+class BaseKubernetesScenario(scenario.Scenario):
+
+    def __init__(self, context=None):
+        super(BaseKubernetesScenario, self).__init__(context)
+        spec = {"namespaces": self.context.get("namespaces"),
+                "serviceaccounts": self.context.get("serviceaccounts")}
+        if "env" in self.context:
+            spec.update(self.context["env"]["platforms"]["kubernetes"])
+            self.client = k8s_service.Kubernetes(
+                spec,
+                name_generator=self.generate_name,
+                atomic_inst=self.atomic_actions())
+
+    def generate_name(self):
+        return self.generate_random_name().replace('_', '-').lower()
```

Comment:
Does kubernetes supports `_` ?

Anyway, it is a wrong thing. It will add a crucial issue while cleanup-ing. The original [generate_random_name](https://github.com/openstack/rally/blob/master/rally/common/utils.py#L334) method doesn't have anything hardcoded. It operates [2 variables](https://github.com/openstack/rally/blob/0.12.0/rally/common/utils.py#L271-L272) which you are interested in. The first one is about format, you can override this variable and change ``_`` by ``-`` (Please note that scenario class and context class has different prefixes. Check base classes for each entity for more details). As for the second variable, it controls which characters should be used while generating the "random" name. As you can see, it doesn't use "upper" chars, so no need in calling ``lower``. But again, if you need to change used chars, it is better to change the special variable.

More explanation: the cleanup mechanism calls [name_matches_object](https://github.com/openstack/rally/blob/0.12.0/rally/common/utils.py#L364) method which operates the same variables. If you change the result of ``generate_random_name`` instead of the method itself, the cleanup will not be able to filter created by Rally resources.

1) Service class accepts generate_random nam